File: android/src/main/kotlin/com/example/pdf/resources/PageRepository.kt
package com.example.pdf.resources

import android.annotation.TargetApi
import android.graphics.pdf.PdfRenderer
import android.os.Build
import com.example.pdf.document.Page
import com.example.pdf.utils.randomID

@TargetApi(Build.VERSION_CODES.LOLLIPOP)
class PageRepository : Repository<Page>() {
    /**
     * Register page in repository
     * @returns page id
     */
    fun register(documentId: String, pageRenderer: PdfRenderer.Page): Page {
        val id = randomID
        val page = Page(id, documentId, pageRenderer)
        set(id, page)
        return page
    }

    public override fun close(id: String) {
        get(id).close()
        super.close(id)
    }
}

File: ios/Classes/resources/PageRepository.swift
//
//  PageRepository.swift
//  pdf
//
//  Created by 林炳午 on 2024/3/8.
//

class PageRepository : Repository<Page> {
    func register(documentId: String, renderer: CGPDFPage) -> Page {
        let id = NSUUID().uuidString
        let page = Page(id: id, documentId: documentId, renderer: renderer)
        set(id: id, item: page)
        return page
    }
}
